clio-client/src/test/scala/org/broadinstitute/clio/client/util/IoUtilSpec.scala

```diff
@@ -1,90 +1,147 @@
-        s"""Hashes [hex] for pipeline/C1963/CHMI_CHMI3_Nex1/v1/id.txt:
-           |        Hash (crc32c):          392eca9a
-           |        Hash (md5):             ${expectedHash.name}
-         """.stripMargin
+
+    (gsutil.hash _).when(uri.toString).returns {
+      s"""Hashes [hex] for pipeline/C1963/CHMI_CHMI3_Nex1/v1/id.txt:
+         |        Hash (crc32c):          392eca9a
+         |        Hash (md5):             ${expectedHash.name}
+       """.stripMargin
     }
 
-    new IoUtil { override val gsUtil: GsUtil = mockGsUtil }
+    new IoUtil { override val gsUtil: GsUtil = gsutil }
       .getMd5HashOfGoogleObject(uri) should be(Some(expectedHash))
   }
 
   it should "not fail when 'gsutil hash' doesn't output an md5 hash" in {
-    val mockGsUtil = new GsUtil {
-      override def hash(path: String): String =
-        s"""Hashes [hex] for pipeline/C1963/CHMI_CHMI3_Nex1/v1/id.txt:
-           |        Hash (crc32c):          392eca9a
-         """.stripMargin
+    val gsutil = stub[GsUtil]
+
+    (gsutil.hash _).when(uri.toString).returns {
+      """Hashes [hex] for pipeline/C1963/CHMI_CHMI3_Nex1/v1/id.txt:
+        |        Hash (crc32c):          392eca9a
+      """.stripMargin
     }
 
-    new IoUtil { override val gsUtil: GsUtil = mockGsUtil }
+    new IoUtil { override val gsUtil: GsUtil = gsutil }
       .getMd5HashOfGoogleObject(uri) should be(None)
   }
 
   it should "parse object size and hash out of 'gsutil stat' output" in {
+    val gsutil = stub[GsUtil]
     val expectedSize = 123412L
     val expectedMd5Hash = Symbol("d41d8cd98f00b204e9800998ecf8427e")
 
-    val mockGsUtil = new GsUtil {
-      override def stat(path: String): String =
-        s"""    Creation time:          Wed, 24 Jan 2018 23:00:14 GMT
-           |    Update time:            Wed, 24 Jan 2018 23:00:14 GMT
-           |    Storage class:          STANDARD
-           |    Content-Length:         $expectedSize
-           |    Content-Type:           application/octet-stream
-           |    Hash (crc32c):          AAAAAA==
-           |    Hash (md5):             1B2M2Y8AsgTpgAmY7PhCfg==
-           |    ETag:                   CKCYzILa8dgCEAE=
-           |    Generation:             1516834814888992
-           |    Metageneration:         1
-         """.stripMargin
+    (gsutil.stat _).when(uri.toString).returns {
+      s"""    Creation time:          Wed, 24 Jan 2018 23:00:14 GMT
+         |    Update time:            Wed, 24 Jan 2018 23:00:14 GMT
+         |    Storage class:          STANDARD
+         |    Content-Length:         $expectedSize
+         |    Content-Type:           application/octet-stream
+         |    Hash (crc32c):          AAAAAA==
+         |    Hash (md5):             1B2M2Y8AsgTpgAmY7PhCfg==
+         |    ETag:                   CKCYzILa8dgCEAE=
+         |    Generation:             1516834814888992
+         |    Metageneration:         1
+       """.stripMargin
```

Comment:
What happens if google changes the output of stat?

Reply:
Our regex will break and this test won't catch it. The proper fix is to stop shelling out to gsutil.